### ISSUE TITLE
raft: on network partition, the term is not incremented on new leader election

### ISSUE THREAD

**Issue by vincentbernat** [benbjohnson/thesecretlivesofdata#17]
Hi!

On the last example, when a partition occurs, the current leader was elected on term 2. The new leader should be elected on term 3. However, it seems to be elected on term 2.

**Comment by naman** [comment]
Yes, I have a similar doubt. How does the minority leader (B) give back the leader position? Is it because it has uncommitted logs (unable to distribute to the majority) or something else entirely.  The vizualization says that (minority leader) B loses the leader position since the cluster's term leader has a higher term value (what does this mean?). I haven't read the paper yet, please forgive me if I am asking a very trivial thing.

**Comment by voteit-top** [comment]
visit raft.github.io, that demo will make you clear.
